### PR TITLE
Support for building and consuming Webpack DLLs

### PR DESCRIPTION
**Pull request by ryandberger**
- Uses new Stripes CLI commands to create and use Webpack DLLs (to be fulfilled by [https://issues.folio.org/browse/STCLI-154](STCLI-154)).
- Allowing `stripesConfig` to be bypassed with `--skipStripesBuild` flag since a DLL (especially third-party) should not perform the additional stripes steps.
- See Stripes CLI PR here: https://github.com/folio-org/stripes-cli/pull/241

### CODE REVIEW

File: webpack/build.js [hunk 1]
```diff
@@ -25,6 +27,34 @@ module.exports = function build(stripesConfig, options) {
...
+      for (let dependency in options.dependencies) {
+        config.plugins.push(new webpack.DllReferencePlugin({
+          context: options.outputPath,
+          manifest: require(path.join(options.outputPath, `${dependency}-manifest.json`))
+        }));
+      }
+    }
+    if (options.createDll || options.useDll) {
+      // When building with Webpack DLL, this alias causes react-dom references to break,
+      // so the remedy is to remove it.
```
[the remark this answers is not in the document]
Comment (ryandberger) [reply webpack/build.js:55]: I'll test it out and remove it if not needed.

File: webpack/build.js [hunk 2]
```diff
@@ -25,6 +27,34 @@ module.exports = function build(stripesConfig, options) {
...
+      config.entry[options.dllName] = options.createDll.split(',');
+      config.output.library = '[name]';
+      config.output.filename = '[name].[hash].js';
+      config.plugins.push(new webpack.DllPlugin({
+        name: '[name]',
+        path: path.join(options.outputPath, '[name]-manifest.json'),
+      }));
+    }
+    if (options.useDll) { // Consume Webpack DLL
+      // Recommended by https://engineering.invisionapp.com/post/optimizing-webpack/ for performance
```
[the remark this answers is not in the document]
Comment (ryandberger) [reply webpack/build.js:41]: Probably! I'll test it out. And good point about the cache options!

File: webpack/build.js [hunk 3]
```diff
@@ -25,6 +27,34 @@ module.exports = function build(stripesConfig, options) {
...
+      config.cache = true;
+      config.devtool = 'eval';
```
[the remark this answers is not in the document]
Comment (ryandberger) [reply webpack/build.js:43]: Yeah... I should check for the option before setting it.

File: webpack/build.js [hunk 4]
```diff
@@ -25,6 +27,34 @@ module.exports = function build(stripesConfig, options) {
...
+      config.dependencies = options.useDll.split(',');
+
+      for (let dependency in options.dependencies) {
+        config.plugins.push(new webpack.DllReferencePlugin({
+          context: options.outputPath,
+          manifest: require(path.join(options.outputPath, `${dependency}-manifest.json`))
```
[the remark this answers is not in the document]
Comment (ryandberger) [reply webpack/build.js:49]: So, I should probably add a new parameter `--dllPath` for this with default to be `outputPath` since that is very convenient for someone creating a DLL and then consuming it in a subsequent command.

File: webpack/build.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ryandberger) [reply webpack/build.js]: Removed for now, since it is general, I will add in a separate PR (as I will with other general cache options).

File: webpack/build.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ryandberger) [reply webpack/build.js]: Not needed. Removed.

File: webpack/build.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ryandberger) [reply webpack/build.js]: Since this is general, will cover in a separate PR.

File: webpack/build.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (ryandberger) [reply webpack/build.js]: Adjusted `--useDll` to support paths.

File: webpack.config.base.js [hunk 5]
```diff
@@ -21,7 +21,6 @@ module.exports = {
   resolve: {
     alias: {
       'react': specificReact,
-      'react-dom': '@hot-loader/react-dom',
```
Comment (skomorokh) [inline webpack.config.base.js:24]: Does this impact the DLL stuff, seems like an unrelated change? Does hot loader still work in dev if you remove this alias?

**Review by skomorokh — APPROVED** [review]
This looks good! Ideally I'd almost rather we just had the DLL usage in build.js and have the DLL creation in a separate dll.js vs. needing to put stuff behind several ifs as it mostly just needs the more static parts eg. webpack.config.* but I can see how that would need the extra wiring to get the stripes-cli options layered in so it's ...probably not worth it.

**Comment by ryandberger** [comment]
> This looks good! Ideally I'd almost rather we just had the DLL usage in build.js and have the DLL creation in a separate dll.js vs. needing to put stuff behind several ifs as it mostly just needs the more static parts eg. webpack.config.* but I can see how that would need the extra wiring to get the stripes-cli options layered in so it's ...probably not worth it.

Yep, I actually started down that route, but there is extra processing too such as handling the `stripes-config` alias which I need to lean on when creating Stripes DLLs.

File: webpack.config.base.js (hunk 5 above)
Comment (ryandberger) [reply webpack.config.base.js:24]: Good question. I'll test it out and if it is needed for dev, I can move it into `webpack.config.cli.dev.js`.

Comment (ryandberger) [reply webpack.config.base.js:24]: Sure enough, without the alias you get this warning in dev mode: `React-Hot-Loader: react-dom patch is not detected. React 16.6+ features may not work.`

I added it back into `webpack.config.cli.dev.js` so it will only apply to dev builds and I confirmed that the warning disappears.

**Review by JohnC-80 — APPROVED** [review]
